mongo/errors.go

```diff
@@ -286,6 +287,30 @@ func (we WriteError) Error() string {
 	return msg
 }
 
+// HasErrorCode returns true if the error has the specified code.
+func (we WriteError) HasErrorCode(code int) bool {
+	return int(we.Code) == code
+}
+
+// HasErrorLabel returns true if the error contains the specified label. WriteErrors do not contain labels,
+// so we always return false.
+func (we WriteError) HasErrorLabel(label string) bool {
+	return false
```

Comment:
As discussed offline `WriteError`s do not contain `Label`s, so we always return false for `HasErrorLabel`.